Extensions/Guardrail+Settings.swift

```diff
@@ -125,12 +125,23 @@ public extension Guardrail where Value == HKQuantity {
 
         } else {
             return Guardrail(
-                absoluteBounds: supportedBasalRates.first!...absoluteUpperBound,
-                recommendedBounds:  supportedBasalRates.first!...absoluteUpperBound,
+                absoluteBounds: supportedBasalRates.drop { $0 <= 0 }.first!...absoluteUpperBound,
+                recommendedBounds:  supportedBasalRates.drop { $0 <= 0 }.first!...absoluteUpperBound,
                 unit: .internationalUnitsPerHour
             )
         }
     }
+    
+    static func selectableBasalRates(supportedBasalRates: [Double],
+                                     scheduledBasalRange: ClosedRange<Double>?,
+                                     lowestCarbRatio: Double?,
+                                     maximumBasalRatePrecision decimalPlaces: Int = 3) -> [Double] {
+        let basalGuardrail = Guardrail.maximumBasalRate(supportedBasalRates: supportedBasalRates, scheduledBasalRange: scheduledBasalRange, lowestCarbRatio: lowestCarbRatio)
+        let maximumScheduledBasalRate = scheduledBasalRange?.upperBound ?? -Double.infinity
```

Comment:
Why is the maximum defaulting to -infinity? I mean, how can a maximum be the smallest possible number? Is this intended to filter out all basal rates? At least a comment should be included to explain why -infinity is used here.

Reply:
This is defaulting to that value in case `scheduledBasalRange` is `nil`.  This effectively makes the `.drop` not "drop" anything in line 142 below.  This logic was copied from DeliveryLimitsEditor.swift, old line 47.

Reply:
I think adding this as a comment would help future readers.

Reply:
I think adding this as a comment would help future readers.